context_chat_backend/models/nc_texttotext.py

```diff
@@ -0,0 +1,83 @@
+import json
+import time
+from typing import Any, Dict, List, Optional
+
+from nc_py_api import Nextcloud
+from langchain_core.callbacks.manager import CallbackManagerForLLMRun
+from langchain_core.language_models.llms import LLM
+
+def get_model_for(model_type: str, model_config: dict):
+	if model_config is None:
+		return None
+
+	if model_type == 'llm':
+		return CustomLLM()
+
+	return None
+
+class CustomLLM(LLM):
+	"""A custom chat model that queries Nextcloud's TextToText provider
+    """
+
+	def _call(
+			self,
+			prompt: str,
+			stop: Optional[List[str]] = None,
+			run_manager: Optional[CallbackManagerForLLMRun] = None,
+			**kwargs: Any,
+	) -> str:
+		"""Run the LLM on the given input.
+
+        Override this method to implement the LLM logic.
+
+        Args:
+            prompt: The prompt to generate from.
+            stop: Stop words to use when generating. Model output is cut off at the
+                first occurrence of any of the stop substrings.
+                If stop tokens are not supported consider raising NotImplementedError.
+            run_manager: Callback manager for the run.
+            **kwargs: Arbitrary additional keyword arguments. These are usually passed
+                to the model provider API call.
+
+        Returns:
+            The model output as a string. Actual completions SHOULD NOT include the prompt.
+        """
+		nc = Nextcloud()
+
+		print(json.dumps(prompt))
+
+		response = nc.ocs("POST", "/ocs/v1.php/taskprocessing/schedule", json={
+			"type": "core:text2text",
+			"appId": "context_chat_backend",
+			"input": {
+				"input": prompt
+			}
+		})
+
+		task_id = response["task"]["id"]
+
+		while response['task']['status'] != 'STATUS_SUCCESSFUL' and response['task']['status'] != 'STATUS_FAILED':
```

Comment:
it would be nice to make a pydantic class here for the response so it can be validated correctly. See https://github.com/nextcloud/context_chat_backend/blob/master/context_chat_backend/controller.py#L277-L299 for ref.

Also, throw a new class of exception like `TaskProcessingException` or `LLMException` when there's any issue in the response object, and make a handler like so for that: https://github.com/nextcloud/context_chat_backend/blob/master/context_chat_backend/controller.py#L84-L87